### ISSUE TITLE
[MNG-4183] CLONE -Maven 2 says host is unavailable while in browser I can download from that repository

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#5958]
**[Adam Saltiel](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=asaltiel)** opened **[MNG-4183](https://issues.apache.org/jira/browse/MNG-4183?redirect=false)** and commented

I am using the offical Glassfish repository on java.net.

Maven tells me that it cannot resolve one of the references pointing into that repository. The stack trace tells me that the host is not available (see below).

In fact, when I go to that same URL in the browser, it downloads the artifact without any problem!

Also that worked for weeks and now is broken, but my POM didn't change!

So I assume a bug in WAGON.

Here is the POM excerpt, below the POM is the stack trace.

    <repositories>
    	<repository>
    		<id>glassfish-repository.dev.java.net</id>
    		<name>Java.net Repository for Glassfish</name>
    		<url>http://download.java.net/maven/glassfish</url>
    	</repository>
    </repositories>
    
    <dependencies>
    	<dependency>
    		<groupId>javax.resource</groupId>
    		<artifactId>connector-api</artifactId>
    		<version>1.5</version>
    		<scope>provided</scope>
    	</dependency>
    </dependencies>

org.apache.maven.lifecycle.LifecycleExecutionException: Unable to get dependency
information: Unable to read the metadata file for artifact 'javax.resource:conn
ector-api:jar': Cannot find parent: org.glassfish.api:api for project: javax.res
ource:connector-api:jar:1.5 for project javax.resource:connector-api:jar:1.5
javax.resource:connector-api:jar:1.5

from the specified remote repositories:
central (http://repo1.maven.org/maven2),
glassfish-repository.dev.java.net (http://download.java.net/maven/glassfish)
Path to dependency:
1) net.sourceforge.rafc:rafc:jar:1.1.1
   ...
   Caused by: java.net.UnknownHostException: download.java.net
   at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:177)
   at java.net.Socket.connect(Socket.java:519)
   at java.net.Socket.connect(Socket.java:469)
   at sun.net.NetworkClient.doConnect(NetworkClient.java:157)
   at sun.net.www.http.HttpClient.openServer(HttpClient.java:394)
   at sun.net.www.http.HttpClient.openServer(HttpClient.java:529)
   at sun.net.www.http.HttpClient.\<init>(HttpClient.java:233)
   at sun.net.www.http.HttpClient.New(HttpClient.java:306)
   at sun.net.www.http.HttpClient.New(HttpClient.java:323)
   at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLC
   onnection.java:788)
   at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConne
   ction.java:729)
   at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection
   .java:654)
   at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLCon
   nection.java:977)
   at org.apache.maven.wagon.providers.http.LightweightHttpWagon.fillInputD
   ata(LightweightHttpWagon.java:83)
   ... 33 more

---

**Affects:** 2.0.8

**Issue Links:**
- [MNG-3373](https://issues.apache.org/jira/browse/MNG-3373) Maven 2 says host in unavailable while in browser I can download from that repository
 (_**"duplicates"**_)

**Comment by jira-importer** [comment]
**[Adam Saltiel](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=asaltiel)** commented

The issue I have is essentially the same as the one I have cloned except that maven is unable to download any artifacts what so ever. Maven version 2.1.0. I am re-opening the issue because, although this error occurs in different forms a lot, I cannot find specifics to a solution in any mailing.
I am unclear what configuration issue might be behind this or if it is a bug.
I have tried a new empty repository and the behaviour is the same.
I do not connect using a proxy and there is no internal firewall to this machine. There is a firewall configured on the terminal adaptor.
However, as with the cloned issue, I can manually download and I know that maven attempts to download. I know this because I tried to setup a proxy server on this machine. Although I am not sure it was correctly set - it worked for my browser at least - the error message from maven changed to access denied.
Below is typical output:-
[DEBUG] Connecting to repository: 'id0' with url: 'http://static.appfuse.org/releases'.                                                                         
Downloading: http://static.appfuse.org/releases/org/appfuse/archetypes/appfuse-basic-tapestry/2.0.2/appfuse-basic-tapestry-2.0.2.jar                            
[DEBUG] Error transferring file: static.appfuse.org                             
org.apache.maven.wagon.TransferFailedException: Error transferring file: static.appfuse.org                                                                     
at org.apache.maven.wagon.providers.http.LightweightHttpWagon.fillInputData(LightweightHttpWagon.java:143)                                              
at org.apache.maven.wagon.StreamWagon.getInputStream(StreamWagon.java:116)                                                                              
at org.apache.maven.wagon.StreamWagon.getIfNewer(StreamWagon.java:88)   
at org.apache.maven.artifact.manager.DefaultWagonManager.getRemoteFile(DefaultWagonManager.java:475)                                                    
at org.apache.maven.artifact.manager.DefaultWagonManager.getArtifact(DefaultWagonManager.java:372)                                                      
at org.apache.maven.artifact.manager.DefaultWagonManager.getArtifact(DefaultWagonManager.java:327)                                                      
at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolve(DefaultArtifactResolver.java:216)                                                 
at org.apache.maven.artifact.resolver.DefaultArtifactResolver.resolveAlways(DefaultArtifactResolver.java:96)                                            
at org.apache.maven.archetype.downloader.DefaultDownloader.downloadOld(DefaultDownloader.java:80)                                                       
at org.apache.maven.archetype.old.DefaultOldArchetype.createArchetype(DefaultOldArchetype.java:122)                                                     
at org.apache.maven.archetype.mojos.MavenArchetypeMojo.execute(MavenArchetypeMojo.java:229)                                                             
at org.apache.maven.plugin.DefaultPluginManager.executeMojo(DefaultPluginManager.java:483)                                                              
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoals(DefaultLifecycleExecutor.java:678)                                                  
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeStandaloneGoal(DefaultLifecycleExecutor.java:553)                                         
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoal(DefaultLifecycleExecutor.java:523)                                                   
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeGoalAndHandleFailures(DefaultLifecycleExecutor.java:371)                                  
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.executeTaskSegments(DefaultLifecycleExecutor.java:268)                                           
at org.apache.maven.lifecycle.DefaultLifecycleExecutor.execute(DefaultLifecycleExecutor.java:181)                                                       
at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:356)       
at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:137)         
at org.apache.maven.cli.MavenCli.main(MavenCli.java:356)                
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)          
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)                                                                        
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)                                                                
at java.lang.reflect.Method.invoke(Method.java:597)                     
at org.codehaus.classworlds.Launcher.launchEnhanced(Launcher.java:315)  
at org.codehaus.classworlds.Launcher.launch(Launcher.java:255)          
at org.codehaus.classworlds.Launcher.mainWithExitCode(Launcher.java:430)
at org.codehaus.classworlds.Launcher.main(Launcher.java:375)            
Caused by: java.net.UnknownHostException: static.appfuse.org                    
at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:177)           
at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:366)           
at java.net.Socket.connect(Socket.java:519)                             
at java.net.Socket.connect(Socket.java:469)                             
at sun.net.NetworkClient.doConnect(NetworkClient.java:163)              
at sun.net.www.http.HttpClient.openServer(HttpClient.java:394)          
at sun.net.www.http.HttpClient.openServer(HttpClient.java:529)          
at sun.net.www.http.HttpClient.\<init>(HttpClient.java:233)              
at sun.net.www.http.HttpClient.New(HttpClient.java:306)                 
at sun.net.www.http.HttpClient.New(HttpClient.java:323)                 
at sun.net.www.protocol.http.HttpURLConnection.getNewHttpClient(HttpURLConnection.java:837)                                                             
at sun.net.www.protocol.http.HttpURLConnection.plainConnect(HttpURLConnection.java:778)                                                                 
at sun.net.www.protocol.http.HttpURLConnection.connect(HttpURLConnection.java:703)                                                                      
at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1026)                                                              
at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:373)                                                                               
at org.apache.maven.wagon.providers.http.LightweightHttpWagon.fillInputData(LightweightHttpWagon.java:115)                                              
... 28 more                                                             
[WARNING] Unable to get resource 'org.appfuse.archetypes:appfuse-basic-tapestry:jar:2.0.2' from repository id0 (http://static.appfuse.org/releases): Error transferring file: static.appfuse.org                                                
[INFO] ----------------------------------------------------------------------------                                                                             
[INFO] Using following parameters for creating OldArchetype: appfuse-basic-tapestry:2.0.2                                                                       
[INFO] ----------------------------------------------------------------------------                                                                             
[INFO] Parameter: groupId, Value: com.mycompany.app                             
[INFO] Parameter: packageName, Value: com.mycompany.app                         
[INFO] Parameter: package, Value: com.mycompany.app                             
[INFO] Parameter: artifactId, Value: myproject                                  
[INFO] Parameter: basedir, Value: /home/adam/workspaces/workspace-03            
[INFO] Parameter: version, Value: 1.0-SNAPSHOT                                  
[INFO] ------------------------------------------------------------------------
[ERROR] BUILD ERROR
